### DISCUSSION TITLE
How does maplibre-gl-js compare to Leaflet and can it display markers and pop-ups?

### DISCUSSION THREAD

**Discussion by fullstackdude** [maplibre/maplibre-gl-js#3026] [Q&A]
Hi,

I am currently using OSM with renderd and on the front end Leaflet.js inside a WebView. Does maplibre-gl-js contain the same type of functionality? If not, how can I get it to work for popups and markers?

Thank you.

**Comment by HarelM** [comment]
MapLibre has more functionality than leaflet. It has markers, popup and more.
I suggest to look at the examples here:
https://maplibre.org/maplibre-gl-js/docs/examples/

**Comment by TimMcCauley** [comment]
If you are using react-leaflet I can also recommend checking out https://visgl.github.io/react-map-gl/examples in combination with MapLibre.
